float32 leftLineInfo
float32 rightLineInfo

